### DISCUSSION TITLE
Allocating array of pointers using new

### DISCUSSION THREAD

**Discussion by tpyra** [ispc/ispc#2623] [General]
Hi

It looks like parser problem.

`int** ip = new int*[100]; // this don't work
`

`int** ip = new (int*)[100]; // this also don't work`

but:
```
typedef int* intp;
int** ip = new intp[100]; // works
```

**Comment by aneshlya** [comment]
```
int** ip = new (int*)[100]; // this also don't work
```
is not valid syntax (it will fail in C++ as well), but
```
int** ip00 = new int*[100]; // this don't work 
```
should work.

Could you create an issue for this in https://github.com/ispc/ispc/issues?

**Reply by tpyra** [reply]
https://github.com/ispc/ispc/issues/2629

**Comment by aneshlya** [comment]
Closing the discussion. The further comments with be in https://github.com/ispc/ispc/issues/2629
